app/src/main/java/com/example/moviedetails/MovieViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.moviedetails
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.moviedetails.network.MovieRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.util.*
+
+class MovieViewModel: ViewModel() {
+
+    private val _listOfMovies = MutableLiveData(listOf(Movie(0, "", "", "", "","")))
+    private val _listOfCurrentYearMovies=MutableLiveData(listOf(Movie(0, "", "", "", "","")))
+
+    val listOfMovies: MutableLiveData<List<Movie>> = _listOfMovies
+    val listOfCurrentYearMovies:LiveData<List<Movie>> = _listOfCurrentYearMovies
+
+    fun getMovies(movieRepository: MovieRepository) {
+        CoroutineScope(Dispatchers.IO).launch {
```

Comment:
you should use viewModelScope. because this scope is constrained to viewmodel lifecycle.

Reply:
Seems like you passed repository in method call. Instead it should be member of Viewmodel